SingularityS3Base/src/main/java/com/hubspot/singularity/s3/base/ArtifactManager.java

```diff
@@ -87,6 +97,37 @@ private void downloadAndCheck(RemoteArtifact artifact, Path downloadTo) {
 
     checkFilesize(artifact, downloadTo);
     checkMd5(artifact, downloadTo);
+    checkGpgSignature(artifact, downloadTo);
+  }
+
+  private void checkGpgSignature(RemoteArtifact baseArtifact, Path downloadTo) {
+    if (baseArtifact.getGpgSignatureArtifact().isPresent() && baseArtifact.getGpgSignatureArtifact().get() instanceof RemoteArtifact) {
+
+      Path signatureArtifactPath =  fetch((RemoteArtifact) baseArtifact.getGpgSignatureArtifact().get());
```

Comment:
Casting is pretty sucky. Why not just add this field to the top-level artifact class if we're going to check it here?